compiler/plugins/target/LLVMCPU/LLVMTargetOptions.cpp

```diff
@@ -588,6 +512,12 @@ void LLVMCPUTargetCLOptions::bindOptions(OptionsBinder &binder) {
       llvm::cl::cat(category),
       llvm::cl::desc("LLVM target machine CPU features; use 'host' for your "
                      "host native CPU."));
+  binder.opt<std::string>(
+      "iree-llvmcpu-logging-unspecified-target-cpu",
+      loggingUnspecifiedTargetCPU, llvm::cl::cat(category),
+      llvm::cl::desc(
+          "Warning behavior on unspecified target CPU. Accepted values: "
+          " (meaning no logging), \"warn\", \"fatal"));
```

Comment:
I would use an enum here if possible, rather than a free-form `std::string`. Pretty sure LLVM has a way to do that: https://llvm.org/docs/CommandLine.html.

Or, and I think I like this more, don't add a flag controlling the logging behavior at all. If we roll this warning/error out and see noisy logs and complaining users, _then_ we can tone down the logging.

There are other performance hints that we might want to log, and I don't like the idea of adding a ~40 character long flag to control the logging for each of them. Maybe we could add a top level verbosity setting for diagnostics (or use whatever upstream may provide: https://mlir.llvm.org/docs/Diagnostics/)

See also
* https://github.com/iree-org/iree/issues/13207